streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -377,6 +377,11 @@ public void closeDirty() {
             streamsProducer.abortTransaction();
         }
 
+        close();
+    }
+
+    private void close() {
+        offsets.clear();
```

Comment:
This is the actual fix.

Reply:
I like having common logic for close clean / close dirty. Should we also move `removeAllProducedSensors` here?

Reply:
I am afraid I do not understand. Method `removeAllProducedSensors()` is only called in `closeClean()` but not in `closedDirty()`.